scene/gui/color_picker.cpp

```diff
@@ -71,18 +60,39 @@ void ColorPicker::_notification(int p_what) {
 				}
 			}
 #endif
-		} break;
-		case NOTIFICATION_PARENTED: {
+			[[fallthrough]];
+		}
+		case NOTIFICATION_THEME_CHANGED: {
+			btn_pick->set_icon(get_theme_icon(SNAME("screen_picker"), SNAME("ColorPicker")));
+			btn_add_preset->set_icon(get_theme_icon(SNAME("add_preset")));
+
+			uv_edit->set_custom_minimum_size(Size2(get_theme_constant(SNAME("sv_width")), get_theme_constant(SNAME("sv_height"))));
+			w_edit->set_custom_minimum_size(Size2(get_theme_constant(SNAME("h_width")), 0));
+
+			wheel_edit->set_custom_minimum_size(Size2(get_theme_constant(SNAME("sv_width")), get_theme_constant(SNAME("sv_height"))));
+			wheel_margin->add_theme_constant_override("margin_bottom", 8 * get_theme_default_base_scale());
+
 			for (int i = 0; i < 4; i++) {
+				labels[i]->set_custom_minimum_size(Size2(get_theme_constant(SNAME("label_width")), 0));
 				set_offset((Side)i, get_offset((Side)i) + get_theme_constant(SNAME("margin")));
 			}
+
+			if (Engine::get_singleton()->is_editor_hint()) {
+				// Adjust for the width of the "Script" icon.
+				text_type->set_custom_minimum_size(Size2(28 * get_theme_default_base_scale(), 0));
```

Comment:
I kept the editor hint condition here because this seems to be aimed specifically at the editor-available icon, but I don't really like this.

Reply:
Yeah it would be good if we could get rid of it. I guess `28` is a magic number of the size of the Script icon?

Looking at the code, this `text_type` seems to be used to allow writing a constructor in the LineEdit, but only in the editor version of the control. In projects, this turns into a flat button which can't be clicked.

It's a bit messy, but also not the only editor-specific hack in this class, so I guess it can be left as is in this PR, and revisited in a future issue/PR.

For example, I'm not sure this feature is actually useful. It was maybe added to allow copy pasting the constructor to use in scripts, but then it might be just as good to have a Copy button for this.

Reply:
I would like to split the color picker up and tidy it at some point, so I can indeed revisit this in a future PR. it's too monstrous at the moment, does too much stuff.